src/esp/assets/ResourceManager.cpp

```diff
@@ -1591,8 +1616,76 @@ int ResourceManager::loadNavMeshVisualization(esp::nav::PathFinder& pathFinder,
   return navMeshPrimitiveID;
 }  // ResourceManager::loadNavMeshVisualization
 
+namespace {
+
+/**
+ * @brief Make sure we do not add an attribute that already exists in source
+ * material - this will assert.
+ * @tparam the type of the value to be added
+ * @param material Source material for new attributes array
+ * @param newAttributes The array of attributes to be used to build a new
+ * Mn::Trade::MaterialData.
+ * @param matAttr The Mn::Trade::MaterialAttribute tag describing the attribute
+ * to add
+ * @param value A reference to the value to add.
+ */
+template <typename T>
+void safeAppend(
```

Comment:
Instead of `safeAppend()`, which doesn't really say *what* is safe about this function, what about `appendIfNotPresent()`?